### PR TITLE
Add a isAjax() method to ServerRequest class

### PR DESCRIPTION
**Pull request by MauricioFauth**
This method belongs to the ServerRequest class instead of the ResponseRenderer class, as it's a request property.

### CODE REVIEW

File: libraries/classes/Http/ServerRequest.php [hunk 1]
```diff
@@ -321,4 +321,10 @@ public function hasBodyParam(string $param): bool
         return is_array($postParams) && isset($postParams[$param])
             || is_object($postParams) && property_exists($postParams, $param);
     }
+
+    public function isAjax(): bool
+    {
+        return $this->serverRequest->getHeaderLine('X-Requested-With') === 'XMLHttpRequest'
+            || $this->has('ajax_request');
```
Comment (kamil-tekiela) [inline libraries/classes/Http/ServerRequest.php:328]: Why is this necessary?

Comment (MauricioFauth) [reply libraries/classes/Http/ServerRequest.php:328]: It's currently used as a way to tell if the response should be HTML or JSON.
